web/autograder_service.go

```diff
@@ -406,17 +406,21 @@ func (s *AutograderService) DeleteGroup(ctx context.Context, in *pb.RecordReques
 }
 
 // GetSubmissions returns the submissions matching the query encoded in the action request.
-// Access policy: Current User if Owner of submission, Teacher of CourseID.
+// Access policy: Current User if Owner of submission (or member of the group in case of group submission), Teacher of CourseID.
 func (s *AutograderService) GetSubmissions(ctx context.Context, in *pb.SubmissionRequest) (*pb.Submissions, error) {
 	usr, err := s.getCurrentUser(ctx)
 	if err != nil {
 		s.logger.Errorf("GetSubmissions failed: authentication error: %w", err)
 		return nil, ErrInvalidUserInfo
 	}
+
+	// even if there is no group ID in request and the grp value is nil, checking whether it contains the current user will just return false
+	grp, _ := s.getGroup(&pb.RecordRequest{ID: in.GetGroupID()})
+
 	// ensure that current user is teacher or the current user is owner of the submission request
 	if !s.hasCourseAccess(usr.GetID(), in.GetCourseID(), func(e *pb.Enrollment) bool {
-		return e.Status == pb.Enrollment_TEACHER || usr.GetIsAdmin() || // TODO: remove isadmin after testing stage
-			(e.Status == pb.Enrollment_STUDENT && (usr.IsOwner(in.GetUserID()) || in.GetGroupID() > 0)) //TODO(meling): The Group check should be more precise; check that usr is in group.
+		return e.Status == pb.Enrollment_TEACHER || usr.GetIsAdmin() ||
```

Comment:
Note that admin is not specified in the access policy.

Reply:
Well, that's a kind of backdoor that you need if you want an admin to be able to check, debug and fix problem for existing course labs by just adding them to the course as a regular student (without promoting to course teacher).
Otherwise you'd have to promote admins to course teachers, and for now this action cannot be undone.

Reply:
ok, then we must add Admin to the policy doc